src/synthesizer_grids/grid_io.py

```diff
@@ -204,7 +205,7 @@ def write_dataset(
         key,
         data,
         description,
-        units="dimensionless",
+        units,
```

Comment:
You need to pass the log_on_read attribute here so that it gets written as an attribute on the dataset itself.